### ISSUE TITLE
option redirect_ssl for FastAPI constructor

### ISSUE THREAD

**Issue by long76** [tiangolo/fastapi#1471]
### Is your feature request related to a problem

Redirect all requests to HTTPS.

### The solution you would like

Set option `redirect_ssl` to `True` and all requests redirect to HTTPS.

**Comment by retnikt** [comment]
ASGI servers are not typically used as the user-facing web servers. It is better to have a reverse proxy such as nginx, Traefik, HAPRoxy, Caddy, or Apache in front of your app which manages the SSL.

**Comment by long76** [comment]
> ASGI servers are not typically used as the user-facing web servers. It is better to have a reverse proxy such as nginx, Traefik, HAPRoxy, Caddy, or Apache in front of your app which manages the SSL.

yeah but Flask support it

**Comment by long76** [comment]
example
```python
@app.before_request
def before_request():
    if request.url.startswith('http://'):
        url = request.url.replace('http://', 'https://', 1)
        code = 301
        return redirect(url, code=code)
```

**Comment by tiangolo** [comment]
Yep, as @retnikt says, it would probably work better in a load balancer like Traefik.

**Comment by long76** [comment]
ok i close issue
